devito/operator.py

```diff
@@ -121,89 +122,62 @@ def __init__(self, expressions, **kwargs):
         # Track the DLE output, as it might be useful at execution time
         self._dle_state = dle_state
 
+        self.symbolic_data = [i for i in parameters if isinstance(i, SymbolicData)]
+        dims = [i for i in parameters if isinstance(i, Dimension)]
+
+        d_parents = [d.parent for d in dims if hasattr(d, 'parent')]
+        self.dims = list(set(dims + d_parents))
+        assert(all(isinstance(i, RuntimeArgProvider)
+                   for i in self.symbolic_data + self.dims))
+        parameters = list(set(parameters + d_parents))
+
         # Finish instantiation
         super(OperatorBasic, self).__init__(self.name, nodes, 'int', parameters, ())
 
-    def arguments(self, *args, **kwargs):
-        """
-        Return the arguments necessary to apply the Operator.
+    def _reset_args(self):
```

Comment:
OK. I was just wondering whether you had in mind to override it at some point, because I saw it was being called from only a single place and it was consisting of only two lines